app/services/shops_list_service.rb

```diff
@@ -1,12 +1,17 @@
 # frozen_string_literal: true
 
 class ShopsListService
+  # shops that are ready for checkout, and have an order cycle that is currently open
   def open_shops
-    shops_list.ready_for_checkout.all
+    shops_list.
+      ready_for_checkout.
+      distributors_with_active_order_cycles
```

Comment:
I noticed in  `active_distributor_ids` we do the filtering with order cycle first and then we filter for ready for check out, extract from the method :
 ```ruby
 enterprises = Enterprise.distributors_with_active_order_cycles.ready_for_checkout
```
Any reason you are doing it the other way around here ? I don't think it matters by maybe one is more efficient than the other ?

Reply:
This wasn't an intential choice, but it shouldn't matter too much when it comes to querying the DB because it only affects the order of some of the `where`s and `join`s - and postgres should be smart enough to plan and execute  both queries in the same way. On the application side, I think any differences would be marginal